src/core/core/api/task.py

```diff
@@ -56,3 +47,30 @@ def initialize(app: Flask):
     task_bp.add_url_rule("", view_func=Task.as_view("tasks"))
     task_bp.add_url_rule("/<string:task_id>", view_func=Task.as_view("task"))
     app.register_blueprint(task_bp)
+
+
+def serialize_result(result: Optional[dict | str] = None):
+    if result is None:
+        return None
+
+    if "exc_message" in result:
+        if isinstance(result["exc_message"], (list, tuple)):
+            return " ".join(result["exc_message"])
+        return result["exc_message"]
+    
+    if "message" in result:
+        return result["message"]
+    return result
+
+
+def handle_task_specific_result(task_id: str, result: Optional[dict | str] = None) -> bool:        
```

Comment:
Is result here still optional? I think if result was None it would not reach this code, but return at line 35